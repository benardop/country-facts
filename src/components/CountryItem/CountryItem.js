import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { getDetials } from '../../redux/detailsReducer/detailsReducer';
import style from './style.module.css';
import spareFlag from '../../assets/images/spareFalg.svg';
import arrow from '../../assets/images/arrow.svg';

function CountryItem(props) {
  const dispath = useDispatch();
  const { country, st } = props;
  const styleObj = {
    backgroundColor: st,
  };
  return (
    <div className={style.container} style={styleObj} data-testid="component-container">
      <Link onClick={() => dispath(getDetials(country.name))} className={style.link} to="/details">
        <img src={arrow} alt="arrow" data-testid="go-to-details" />
      </Link>

      <div>
        <img
          src={country.flag}
          alt="flag"
          onError={({ currentTarget }) => {
            const image = currentTarget;
            image.onerror = null;
            image.src = spareFlag;
          }}
        />
      </div>
      <h1>{country.name}</h1>
      <h2>
        {country.timezone}
      </h2>
    </div>
  );
}

CountryItem.propTypes = {
  country: PropTypes.objectOf(PropTypes.string).isRequired,
  st: PropTypes.string.isRequired,
};

export default CountryItem;
