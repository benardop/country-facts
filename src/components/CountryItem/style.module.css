@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&family=Puritan&display=swap');

.container {
  display: flex;
  position: relative;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  box-sizing: border-box;
  flex-basis: 30%;
}

.container h1 {
  font-size: 1.5rem;
  font-family: 'Puritan', sans-serif;
  color: white;
}

.link {
  position: absolute;
  top: 1rem;
  right: 1rem;
}

.link img {
  width: 2rem;
}

.container div {
  height: 200px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.container div img {
  width: 90%;
  max-height: 200px;
}

.container h2 {
  font-size: medium;
  font-family: 'Lato', sans-serif;
  color: white;
}

@media only screen and (max-width: 700px) {
  .container {
    flex-basis: 90%;
  }
}
