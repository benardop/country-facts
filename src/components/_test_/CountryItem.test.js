import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import CountryItem from '../CountryItem/CountryItem';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if Item component has changed', () => {
  const country = {
    name: 'Iraq',
    capital: 'Bagdad',
    flag: 'https//flag',
    timezone: 'utc+1',
  };
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <CountryItem country={country} />
        ,
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Check if the component container is there', async () => {
  const country = {
    name: 'Iraq',
    capital: 'Bagdad',
    flag: 'https//flag',
    timezone: 'utc+1',
  };
  render(
    <Provider store={store}>
      <BrowserRouter>
        <CountryItem country={country} st="red" />
      </BrowserRouter>
    </Provider>,
  );
  const container = await screen.findByTestId('component-container');
  expect(container).toBeInTheDocument();
});
