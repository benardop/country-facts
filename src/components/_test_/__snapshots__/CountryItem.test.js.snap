// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Item component has changed 1`] = `
Array [
  <div
    className="container"
    data-testid="component-container"
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
  >
    <a
      className="link"
      href="/details"
      onClick={[Function]}
    >
      <img
        alt="arrow"
        data-testid="go-to-details"
        src="arrow.svg"
      />
    </a>
    <div>
      <img
        alt="flag"
        onError={[Function]}
        src="https//flag"
      />
    </div>
    <h1>
      Iraq
    </h1>
    <h2>
      utc+1
    </h2>
  </div>,
  ",",
]
`;
